src/server/pfs/cmds/mount_unix.go

```diff
@@ -15,27 +15,58 @@ import (
 	"github.com/pachyderm/pachyderm/src/server/pfs/fuse"
 	"github.com/pachyderm/pachyderm/src/server/pkg/cmdutil"
 
-	"github.com/hanwen/go-fuse/fuse/nodefs"
+	"github.com/hanwen/go-fuse/v2/fs"
+	gofuse "github.com/hanwen/go-fuse/v2/fuse"
 	"github.com/spf13/cobra"
 )
 
-func parseCommits(args []string) (map[string]string, error) {
-	result := make(map[string]string)
+func parseRepoOpts(args []string) (map[string]*fuse.RepoOptions, error) {
+	result := make(map[string]*fuse.RepoOptions)
 	for _, arg := range args {
-		split := strings.Split(arg, "@")
-		if len(split) != 2 {
-			return nil, errors.Errorf("malformed input %s, must be of the form repo@commit", args)
+		var repo string
+		var flag string
+		opts := &fuse.RepoOptions{}
+		repoAndRest := strings.Split(arg, "@")
+		if len(repoAndRest) == 1 {
+			// No branch specified
+			opts.Branch = "master"
+			repoAndFlag := strings.Split(repoAndRest[0], "+")
+			repo = repoAndFlag[0]
+			if len(repoAndFlag) > 1 {
+				flag = repoAndFlag[1]
+			}
+		} else {
+			repo = repoAndRest[0]
+			branchAndFlag := strings.Split(repoAndRest[1], "+")
+			opts.Branch = branchAndFlag[0]
+			if len(branchAndFlag) > 1 {
+				flag = branchAndFlag[1]
+			}
+		}
+		if flag != "" {
+			for _, c := range flag {
+				if c != 'w' && c != 'r' {
+					return nil, errors.Errorf("invalid format %q: unrecognized mode: %q", arg, flag)
```

Comment:
I think this error should report the unrecognized mode character rather than the entire mode.
```suggestion
					return nil, errors.Errorf("invalid format %q: unrecognized mode: %q", arg, c)
```